app/src/main/java/io/github/hidroh/materialistic/CustomTabsDelegate.java

```diff
@@ -110,8 +112,36 @@ void onServiceDisconnected() {
     }
 
     private static String getPackageNameToUse(Context context) {
-        // packagesSupportingCustomTabs contains all apps that can handle both VIEW intents
-        // and service calls.
+        List<String> browserWithCustomTabsSupport = getBrowserWithCustomTabsSupport(context);
+        String defaultBrowser = getDefaultBrowser(context);
+
+        for (String browser : browserWithCustomTabsSupport) {
+            if (browser.equals(defaultBrowser)) {
+                return browser;
+            }
+        }
+
+        return browserWithCustomTabsSupport.get(0);
+    }
+
+    /**
+     * Returns the package name of the default browser on the device
+     * @param context Context
+     * @return The package name of the default browser
+     */
+    private static String getDefaultBrowser(Context context) {
```

Comment:
Can you annotate this `@Nullable` since it could be null if no browsers install?

Reply:
Done